File: two_pointers/sum_of_variable_length_subarrays.py
class Solution:
    def subarraySum(self, nums: List[int]) -> int:
        total_sum = 0
        
        for i, num in enumerate(nums):
            start = max(0, i - nums[i])
            total_sum += sum(nums[start:i + 1])
            
        return total_sum

File: two_pointers/sum_of_variable_length_subarrays.rb
# @param {Integer[]} nums
# @return {Integer}
def subarray_sum(nums)
    sum = 0
    
    nums.each_with_index do |num, index|
        start = [0,(index - (nums[index]))].max
        sum += nums[start..index].sum
    end
    return sum
end